### PR TITLE
Adding rich push setup for demo

### PR DESCRIPTION
**Pull request by ajaysubra**
# Description

Since we didn't have a demo example for rich push and a developer reported having some issues with it, adding an example here. 

# Check List

- [ ] Are you changing anything with the public API?
- [ ] Are your changes backwards compatible with previous SDK Versions?

### CODE REVIEW

File: example/ios/KlaviyoReactNativeSdkExampleExtension/NotificationService.swift [hunk 1]
```diff
@@ -0,0 +1,53 @@
+//
```
Comment (evan-masseau) [inline example/ios/KlaviyoReactNativeSdkExampleExtension/NotificationService.swift:1]: In other places, you had included a comment like `Setup Step #`, thought that was nice for searchability. Would it make sense to label this file, or a particular spot, with that same comment string?

File: example/ios/KlaviyoReactNativeSdkExampleExtension/NotificationService.swift [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+import KlaviyoSwiftExtension
+import UIKit
+import UserNotifications
+
+// MARK: notification service extension implementation.
+
+/// When push payload is marked as there being mutable-content this service
+/// (more specifically the `didReceiveNotificationRequest` ) is called to perform
```
Comment (evan-masseau) [inline example/ios/KlaviyoReactNativeSdkExampleExtension/NotificationService.swift:15]: super nit
```suggestion
/// (more specifically `didReceiveNotificationRequest` ) is called to perform
```

**Review by evan-masseau — APPROVED** [review]
works for me! left some non-blocking comments inline, and one additional tiny suggestion since you're looking at example app:

File: example/ios/KlaviyoReactNativeSdkExampleExtension/NotificationService.swift [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+import UIKit
+import UserNotifications
+
+// MARK: notification service extension implementation.
+
+/// When push payload is marked as there being mutable-content this service
+/// (more specifically `didReceiveNotificationRequest` ) is called to perform
+/// tasks such as downloading images and attaching it to the notification before it's displayed to the user.
+///
+/// There is a limited time before which `didReceiveNotificationRequest`  needs to wrap up it's operations
```
Comment (jordan-griffin) [inline example/ios/KlaviyoReactNativeSdkExampleExtension/NotificationService.swift:18]: nit: `There is limited time`, the `a` in this sentence looks unnecessary

File: example/ios/KlaviyoReactNativeSdkExampleExtension/NotificationService.swift [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+//  Created by Ajay Subramanya on 4/9/24.
+//
+
+import KlaviyoSwiftExtension
+import UIKit
+import UserNotifications
+
+// MARK: notification service extension implementation.
+
+/// When push payload is marked as there being mutable-content this service
```
Comment (jordan-griffin) [inline example/ios/KlaviyoReactNativeSdkExampleExtension/NotificationService.swift:14]: total nit but: `When a push payload` sounds a little better

Comment (jordan-griffin) [reply example/ios/KlaviyoReactNativeSdkExampleExtension/NotificationService.swift:14]: Or maybe even `When a push payload is marked as having mutable-content this service`...

Comment (ajaysubra) [reply example/ios/KlaviyoReactNativeSdkExampleExtension/NotificationService.swift:14]: I removed this as this is sample code from iOS repo and probably not required here. Will fix it there in a different PR.

File: example/ios/KlaviyoReactNativeSdkExampleExtension/NotificationService.swift (hunk 3 above)
Comment (ajaysubra) [reply example/ios/KlaviyoReactNativeSdkExampleExtension/NotificationService.swift:18]: Same remove those comments.

**Pull request merged**